### DISCUSSION TITLE
Error about yy annihilation in your blog post "Interaction Nets, Combinators, and Calculus" posted March 26, 2023

### DISCUSSION THREAD

**Discussion by DiegoCabreraPell** [zicklag/blog#6] [General]
It appears that the yy (constructor) annihilation result in your post is incorrect. As far as I have found the rule should not create the same cross pattern as dd (duplicator) but instead two straight wirings. I have linked a couple of sources including a paper from the inventor of Interaction nets:

https://www.sciencedirect.com/science/article/pii/S0890540197926432 page 13
https://wiki.xxiivv.com/site/interaction_nets.html In the programming section

Link to your post for convenience:
https://zicklag.katharos.group/blog/interaction-nets-combinators-calculus/

Otherwise, I found the post to be very helpful for understanding how lambda calculus can be encoded into interaction combinators and genuinely a fun read!

**Comment by zicklag** [comment]
I've spent some time away from this and I can't get into the weeds again right now, but I _think_ that was a difference from the original paper, implemented by the [Interaction Calculus](https://github.com/victortaelin/interaction-calculus?tab=readme-ov-file#why) from [HVM](https://github.com/HigherOrderCO/HVM), which is the formula that I was following.

I think HVM extended the idea of constructors and duplicators more generically to "nodes" with a "tag", and if it had the same tag it would perform an annihilation and if it was a different tag, it would perform a duplication.

Also, I'm super glad you liked the post. :smile:

**Reply by zicklag** [reply]
I've only very quickly skimmed it, but this paper might explain it:

https://www.lipn.univ-paris13.fr/~mazza/papers/CombSem-MSCS.pdf

I think it's what they are calling Symmetric Interaction Combinators, as distinguished from Interaction Combinators by Yves Lafont.

**Reply by DiegoCabreraPell** [reply]
I see, sorry for the misunderstanding and thank you for the quick reply! I will certainly read the paper, it looks very interesting.

P.S. Looking forward to the next blog post!

**Reply by zicklag** [reply]
No problem at all!

Unfortunately all this interaction-combinator stuff was a bit of a side-quest for me, and I ended up running out of time for it as more important things came up, so I might not get to writing anything else on the topic for a good while, if ever.

Still, you never know what might happen. I am still interested in it, and want to get to it some day. :)
